Auth_phpbb.php

```diff
@@ -504,15 +590,50 @@ public function getCanonicalName( $username )
 
         // Query Database.
         $fresStatement = $fresMySQLConnection->prepare($fstrMySQLQuery);
-        $fresStatement->bind_param('s', $username);
+        $fresStatement->bind_param('s', $this->_phpBBUserName); // bind_param escapes the string
         $fresStatement->execute();
 
         // Bind result
-        $fresStatement->bind_result($resultUsernameClean);
+        $fresStatement->bind_result($resultWikiUsername);
 
         while($fresStatement->fetch())
         {
-            return ucfirst($resultUsernameClean);
+            $this->_wikiUserName = ucfirst($resultWikiUsername); // Preserve capped phpBB username when wikified version is valid
+            return $this->_wikiUserName;
+        }
+
+        // If here, username is invalid or is incompatible with wiki username.
```

Comment:
When I say user I mean user of this extension, i.e. the wiki/forum admin, not the users of those sites. We can't expect site admins who use this extension to be experts on it, and they *are* going to be confused by this.

I'm not suggesting that site end users should be allowed to set aliases, I'm saying that the site admins should be allowed to decide how to use this feature on their own, without restrictions beyond their control.